subprojects/core/src/main/java/org/gradle/api/attributes/Attribute.java

```diff
@@ -117,4 +119,30 @@ public int hashCode() {
     public String toString() {
         return name;
     }
+
+    private static <T> void validateType(String name, Class<T> clazz) {
+        if (clazz.isArray()) {
+            validateType(name, clazz.getComponentType());
+            return;
+        }
+        if (Scalars.isScalarType(clazz)) {
+            return;
+        }
+        if (Named.class.isAssignableFrom(clazz)) {
```

Comment:
That's true. However strictly speaking, the ticket is about types which _value we can make immutable_. The type itself doesn't have to. For named, it would be a lossy conversion, because we would only keep the name.

The thing is that you declare a type like this:

```
interface Flavor implements Named {}
```

and only this. So one option is just to check that the type in question is an interface type. So `Named` **and** interface. This would make it in practice impossible for the user to use methods for which we don't know how to generate an implementation.